mac/focal yaml serialization differs from bionic's

For whatever reason, yaml serialization on mac and focal produces vertically formatted lists, while serialization on bionic produces horizontally formatted ones.  This produces errors like

```AssertionError: 'exam[21 chars]rams:\n  - 5.0\n  - 50.0\n  - 5.0\n  - 1000.0\[89 chars]05\n' != 'exam[21 chars]rams: [5.0, 50.0, 5.0, 1000.0]\n  initial_stat[59 chars]05\n'
  example:
+   controller_params: [5.0, 50.0, 5.0, 1000.0]
+   initial_state: [1.2, 0.0, 0.0, 0.0]
-   controller_params:
-   - 5.0
-   - 50.0
-   - 5.0
-   - 1000.0
-   initial_state:
-   - 1.2
-   - 0.0
-   - 0.0
-   - 0.0
    t_final: 30.0
    tape_period: 0.05
```

The relevant test was nerfed in #13538 but should be rewritten to compare the parsed (or post-parse serialized) rather than a raw reference serialized form in its roundtrip tests (ie, to insist on semantic rather than syntactic equality).

--- a/examples/acrobot/dev/BUILD.bazel
+++ b/examples/acrobot/dev/BUILD.bazel
@@ -43,12 +43,6 @@ drake_py_unittest(
     data = [
         ":test/example_scenario.yaml",
     ],
-    tags = [
-        # TODO(jwnimmer-tri) This test is disabled because it fails on macOS
-        # due to formatting differences.  We should make it more robust and
-        # then re-enable it.
-        "manual",
-    ],
     deps = [
         ":acrobot_io",
         "//bindings/pydrake",

--- a/examples/acrobot/dev/acrobot_io.py
+++ b/examples/acrobot/dev/acrobot_io.py
@@ -2,6 +2,12 @@ import yaml
 
 import numpy as np
 
+# This is a magic tribool value described at
+#   https://github.com/yaml/pyyaml/pull/256
+# and must be set this way for post- and pre- pyyaml#256 yaml dumpers to give
+# the same output.
+_FLOW_STYLE = None
+
 
 def load_scenario(*, filename=None, data=None, scenario_name=None):
     """Given a scenario `filename` xor `data`, and optionally `scenario_name`,
@@ -42,7 +48,9 @@ def save_scenario(*, scenario, scenario_name):
                 ]
         else:
             scrubbed[key] = [float(x) for x in scenario[key]]
-    return yaml.dump({scenario_name: scrubbed}, Dumper=yaml.CDumper)
+    return yaml.dump({scenario_name: scrubbed},
+                     Dumper=yaml.CDumper,
+                     default_flow_style=_FLOW_STYLE)
 
 
 def load_output(*, filename=None, data=None):
@@ -68,4 +76,5 @@ def load_output(*, filename=None, data=None):
 def save_output(*, x_tape):
     """Given an acrobot output `x_tape`, returns a yaml-formatter str for it.
     """
-    return yaml.dump({"x_tape": x_tape.tolist()})
+    return yaml.dump({"x_tape": x_tape.tolist()},
+                     default_flow_style=_FLOW_STYLE)

